### DISCUSSION TITLE
Performance Issue: Slow Endpoint Response Time in Node.js Backend on Vercel

### DISCUSSION THREAD

**Discussion by Arthur-AOB** [community/community#142205] [Programming Help] [answered]
### Body

I developed a Node.js backend, but I’m encountering a significant performance issue. One of my endpoints is taking over one minute to respond, which is problematic because Vercel has a default limit of 10 seconds for HTTP requests.

**Problem Details:**

**Environment**: Node.js backend deployed on Vercel
**Issue**: One endpoint is slow (over 1 minute response time)
**Impact**: The endpoint cannot be used on Vercel due to the 10-second request limit

I would appreciate any insights, suggestions, or experiences from the community regarding to find the easiest way to solve this issue. Has anyone faced similar challenges on Vercel, and what solutions worked for you? Thank you!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by hardcarrier** [comment]
Did you try to upgrade the Vercel plan?

**Reply by Arthur-AOB** [reply]
That won't be the solution. I need programmatical solutions.

**Comment by sven3270350** [comment, marked as answer]
To resolve this issue, you can implement a polling API to check the completion status of the long-running process.

1. **Base API Controller**: In your main API controller, log the request details in the database immediately and return a simple response containing the record ID. This way, you avoid any waiting time on the front end.

2. **Polling API**: Create a separate polling API that checks the database for updates based on the record ID provided in the initial response. 

3. **Frontend Implementation**: On the front end, call the polling API every 3 to 5 seconds to check the status. Once the polling API indicates that the process is complete, you can then call the getResult API to retrieve the actual response.

This approach allows for efficient handling of long-running tasks without exceeding the timeout limits.

**Reply by Arthur-AOB** [reply]
Totally, Should I develop 2 additional endpoints?

**Reply by sven3270350** [reply]
Yeah, sure.
Also needs one more table to track the API details.

**Reply by Arthur-AOB** [reply]
It's possible I think.
Let me try this on my side and let you know if I have any blocker.
Really thanks for your reply.
